### ISSUE TITLE
Memory increases during training epoch, limiting training set size.

### ISSUE THREAD

**Issue by nkemnitz** [ZettaAI/zetta_utils#543]
Memory keeps increasing during a single training epoch, after which it drops back to normal levels.
Not sure if bug or intented behavior, but it meant I had to split my training set into smaller chunks and rotate them in and out every "epoch".

I removed all the wandb logging code (especially the per-epoch logging flags) from my regime, but no luck with that.

**Comment by supersergiy** [comment]
@nkemnitz did we ever resolve this?

**Comment by nkemnitz** [comment]
I don't think so, I haven't trained anything else since then, and we don't have any other datasets of this size, I think? @ranlu pointed me to https://ppwwyyxx.com/blog/2022/Demystify-RAM-Usage-in-Multiprocess-DataLoader/ earlier this month.
